### ISSUE TITLE
Using dMaSIF-site for interaction prediction

### ISSUE THREAD

**Issue by DavidGraber** [casperg92/MaSIF_colab#3]
Hi!

Thanks to the authors for this great notebook! I'm trying to use dMaSIF for interaction prediction between proteins (taking a target and finding the best binder in a large collection of potential binders)

At the moment, I process both binder and target molecule identically with dMaSIF-site, and I export the outputs "xxxx_predfeatures_emb1.npy" and "predcoords.npy". From the feature files I'm taking columns 17-32 (which I believe contain the 10 geometric and 6 learned chemical features after some convolutions, MLPs and batch normalizations). 

According to the paper, these features of both binding partners are then passed through a separate convolutional network, allowing the network to find **complementary** (instead of similar) regions. Unfortunately I was not able to find the code doing that. Could you point me to the right section in the dMaSIF code?

Thanks so much to all contributors
DavidGraber

**Comment by casperg92** [comment]
Hi David,

Yes this is correct. Currently, we only have an implementation of dMaSIF-site available, the complementarity fingerprints you're referring to is dMaSIF-search. We're still working on making a dMaSIF-search implementation for google colab. 

The problem here is that dMaSIF-search only gives you a score for the complementarity between two patches (aka two individual points), however, an interaction exists out of a patch of these points (so a patch of patches). We're still looking for a good way to score these interactions. As soon as we've found this we'll update the repo and let you know.

Best,

Casper

**Comment by DavidGraber** [comment]
Hi Casper

Thanks so much for your response! That's exactly what I want, to extract patches from both proteins and score the complementarity of these patches. By patches I mean a collection of surface points (with their features) taken from a circle of fixed geodesic radius. 

I'm looking forward to the dMaSIF-search Colab notebook! Meanwhile, I have downloaded the dMasif code (non-colab) from Freyr Sverrisons GitHub. In this code, there should be code that extracts patches from the two binding partners and passes them through a separate convolutional network to score their complementarity, shouldn't it?

Unfortunately, I'm not able to find that code. I would be very grateful if you could help me with that!

Best, 
David

**Comment by casperg92** [comment]
Hi David,

To add to Arnes reply, you can find the function to calculate the complementarity in the loss function here:
https://github.com/FreyrS/dMaSIF/blob/0dcc26c3c218a39d5fe26beb2e788b95fb028896/data_iteration.py#L157

Best,

Casper
